### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by najibullahjafari**
Here:

- I implemented the design from the sneak peek wireframes.
now it looks like this:
Screenshot 2023-10-13 041632

- I used methods that I have created in the [Blog App - processing data in models]

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class UsersController < ApplicationController
   def index
     @users = User.all
-    render json: @users
+    render index: @users
   end
```
Comment (Shedrack-Sunday) [inline app/controllers/users_controller.rb:5]: - [ ] Your `user` views is missing a couple of methods which was meant to be implemented in the previous milestone, as they will be needed in this milestone. One of such method is the `#recent_post` method to render the user's 3 most recent posts. Another is the `comments_counter` and also `likes_counter`. This methods will help you render the views properly 👍 

### Requirement:
image

### Link to methods you're missing: [Here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md)
image


### Sample of expected code to show users:
```
# app/views/users/show.html.erb

[...]

<% @user.recent_post.each do |post| %>

<h4><%= link_to post.title, user_post_path(post) %></h4>
<p><%= post.text %></p>
<p>
<small>Comments: <%= post.comments_counter || 0 %>,</small> 
<small>Likes: <%= post.likes_counter || 0 %></small>
</p>

<% end %>

<button>
<%= link_to 'See all Post', user_posts_path(@user) %>
</button>

```

File: app/views/posts/show.html.erb [hunk 2]
```diff
@@ -1,4 +1,24 @@
...
-<p><%= @post.text %></p>
-<p>Comments: <%= @post.comments_counter %></p>
-<p>Likes: <%= @post.likes_counter %></p>
+<div class="post-details">
+  <h2 class="mb-0"><%= @post.title %></h2>
+  <p class="text-muted mb-2">Posted by <strong><%= @post.author.name %></strong></p>
+  <p class="text-muted">
+    <span class="me-3">comment count: <strong><%= @post.comments_counter %></strong></span>
+    like count: <strong><%= @post.likes_counter %></strong>
+  </p>
```
Comment (Shedrack-Sunday) [inline app/views/posts/show.html.erb]: - [ ] Note: in the `app/views/posts/index.html.erb` file, you're expected to uses `post.recent_comments` to render 5 most recent comments. This method is also missing as its part of the previous [milestone](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md)

### Sample of expected code:
```
 <% post.recent_comments.each do |comment| %>
  <p><%= comment.author.name %>: <%= comment.text %></p> 
 <% end %>
```

**Review by Shedrack-Sunday — CHANGES_REQUESTED** [review]
Hello @najibullahjafari          👋 

# CHANGES REQUIRED ♻️

Impressive work on the project so far👏
You've done extremely well so far in this project,  However, there are some adjustments you need to make to move to the next project. 💯



## Highlights   
- Good use of GitHub features ✔️   
- Nice work with the Rails features ✔️
- Good job with view templates    ✔️  

# - Required Changes  🌵  

- [x] Though you made some changes as requested in this milestone, there are a couple of methods needed in this milestone that were in the previous milestone but not implemented. 
- I gave a very comprehensive review in the comments, so you can implement them can meet the requirements as stated


_Check the comment below




# Optional suggestions 

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing really improves your application.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (najibullahjafari) [reply app/controllers/users_controller.rb:5]: @Shedrack-Sunday 
I already have all the methods in the models!

Comment (Shedrack-Sunday) [reply app/controllers/users_controller.rb:5]: I saw them. Please implement them in your views as I snapped in the requirements

**Comment by najibullahjafari** [comment]
Screenshot 2023-10-13 090854
Screenshot 2023-10-13 090903
Screenshot 2023-10-13 090909

**Review by topeogunleye — APPROVED** [review]
### Hi @najibullahjafari, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 

### Implemented User View Methods: 🧠🔄
- **Recent Posts Method**: Adding the `#recent_post` method to retrieve the user's 3 most recent posts shows great adherence to the requirements and will undoubtedly enhance the user experience by spotlighting fresh content. Bravo on swiftly incorporating this! 🌟📜
  
- **Comments and Likes Counter**: Introducing the `comments_counter` and `likes_counter` methods demonstrates a thoughtful approach towards providing relevant and quick insights on posts, directly in the user view. This is pivotal for a dynamic and interactive user interface! 💬👍

### Handling Post Comments: 💬🔄
- Adding the mechanism to render the 5 most recent comments with `post.recent_comments` in the `app/views/posts/index.html.erb` file adds an extra layer of dynamism and user engagement in your application. Engaging with recent discussions is a vital part of social applications, and your implementation caters well to that. Fantastic job! 🎉🗣️

### Summary: 📘🌐
Your keenness in addressing feedback and making necessary adjustments is not only commendable but also instrumental in evolving your coding journey. The implementations are spot on and tie nicely with the project requirements, ensuring that users are always interacting with recent and relevant data.

Here's an encouragement to always validate your methods by writing tests to ensure they're always returning the expected outputs in all scenarios. 🧪🛠️ 

Continue marching forward with this positive momentum, and remember, the journey of code refinement is infinite, embracing constant learning and improvement! 🌱🔄

If you have further queries or face any hurdles, do not hesitate to reach out. Happy coding! 🖥️❤️

Warm regards,
Temitope
 
## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**